ckks/plaintext.go

```diff
@@ -26,3 +27,12 @@ func (p *Plaintext) ScalingFactor() float64 {
 func (p *Plaintext) SetScalingFactor(scale float64) {
 	p.Scale = scale
 }
+
+// NewPlaintextAtLevelFromPoly construct a plaintext at a specific level
+// from a polynomial, without modifying the polynomial.
+func NewPlaintextAtLevelFromPoly(level int, poly *ring.Poly) *Plaintext {
+	v0 := new(ring.Poly)
+	v0.IsNTT = true
+	v0.Coeffs = poly.Coeffs[:level+1]
+	return &Plaintext{Plaintext: &rlwe.Plaintext{Value: v0}, Scale: 0}
```

Comment:
Godoc must be explicit that the returned plaintext shares its backing coefficient array with `poly`.